crates/services/sync/src/import.rs

```diff
@@ -302,6 +315,72 @@ fn validate_header_height(
     header.entity.consensus.height == expected_height
 }
 
+#[tracing::instrument(
+    skip(p2p, header),
+    fields(
+        height = **header.entity.height(),
+        id = %header.entity.consensus.generated.application_hash
+    ),
+    err
+)]
+async fn get_transactions_on_block<P>(
+    p2p: &P,
+    block_id: SourcePeer<BlockId>,
+    header: SealedBlockHeader,
+) -> anyhow::Result<Option<SealedBlock>>
+where
+    P: PeerToPeerPort + Send + Sync + 'static,
+{
+    let Sealed {
+        entity: header,
+        consensus,
+    } = header;
+
+    // Request the transactions for this block.
+    Ok(p2p
+        .get_transactions(block_id)
+        .await
+        .trace_err("Failed to get transactions")?
+        .trace_none_warn("Could not find transactions for header")
+        .and_then(|transactions| {
+            let block = Block::try_from_executed(header, transactions)
+                .trace_none_warn("Failed to created header from executed transactions")?;
+            Some(SealedBlock {
+                entity: block,
+                consensus,
+            })
+        }))
+}
+
+#[tracing::instrument(
+    skip_all,
+    fields(
+        height = **block.entity.header().height(),
+        id = %block.entity.header().consensus.generated.application_hash
+    ),
+    err
+)]
+async fn execute_and_commit<E>(
+    executor: &E,
+    state: &SharedMutex<State>,
+    block: SealedBlock,
+) -> anyhow::Result<()>
+where
+    E: BlockImporterPort + Send + Sync + 'static,
+{
+    // Execute and commit the block.
+    let height = *block.entity.header().height();
+    let r = executor.execute_and_commit(block).await;
+
+    // If the block executed successfully, mark it as committed.
+    if r.is_ok() {
+        state.apply(|s| s.commit(*height))
```

Comment:
we should info-level log here to indicate sync progress

Reply:
Although it seems like maybe this should be logged already via `tracing::instrument`? I'm not seeing this fn in the logs when I enable `FUEL_TRACE=1`

Reply:
was able to get instrument to show up using:

```
            "1" | "true" | "on" => {
                let _ = tracing_subscriber::FmtSubscriber::builder()
                    .with_env_filter(EnvFilter::from_default_env())
                    .with_span_events(FmtSpan::NEW | FmtSpan::CLOSE)
                    .try_init();
            }
```

However it is _very_ noisy, so I still think more info! events could be useful.

Reply:
It's logged in the state.rs but it's debug level. I can bump it up though